### PR TITLE
Enemies of the state/revolution cannot be defibbed

### PR DESCRIPTION
**Pull request by JohnFulpWillard**
## About The Pull Request

image

## Why It's Good For The Game

This is an oversight, which made them just unable to be podcloned, despite the centcom announcement (for headrevs, but command/sec are also made unclonable so I assume it is the same for them), saying they are 'blacklisted from medical records to ensure no revival'.

## Changelog

:cl:
fix: Enemies of the State and (Command-only) Enemies of the Revolution are now ACTUALLY blacklisted from being defibrilated.
/:cl:

### CODE REVIEW

**Review by Mothblocks — CHANGES_REQUESTED** [review]
1. Make sure that if a head of staff leaves the station, comes back, and dies, they are unrevivable. You'll need a trait or similar. Don't do this on the antag datum (like you pitched on Discord)
2. Security should be revivable, just not heads.

File: code/__DEFINES/traits.dm [hunk 1]
```diff
@@ -147,6 +147,7 @@ Remember to update _globalvars/traits.dm if you're adding/removing/renaming trai
 #define TRAIT_NEARSIGHT "nearsighted"
 #define TRAIT_FAT "fat"
 #define TRAIT_HUSK "husk"
+#define TRAIT_DEFIB_BLACKLISTED "defib_blacklisted"
```
Comment (Mothblocks) [inline code/__DEFINES/traits.dm:150]: Doc-comment it

File: code/modules/antagonists/revolution/revolution.dm [hunk 2]
```diff
@@ -441,39 +441,34 @@
...
 		// If the revolution was quelled, make rev heads unable to be revived through pods
-		for (var/_rev_head_mind in ex_revs)
-			var/datum/mind/rev_head_mind = _rev_head_mind
+		for (var/datum/mind/rev_head_mind as anything in ex_revs)
 			var/mob/living/carbon/rev_head_body = rev_head_mind.current
-			if(istype(rev_head_body) && rev_head_body.stat == DEAD)
-				rev_head_body.makeUncloneable()
+			if(!istype(rev_head_body) || rev_head_body.stat != DEAD)
+				continue
+			ADD_TRAIT(rev_head_body, TRAIT_DEFIB_BLACKLISTED, JOB_TRAIT)
```
Comment (Mothblocks) [inline code/modules/antagonists/revolution/revolution.dm:448]:
```suggestion
			ADD_TRAIT(rev_head_body, TRAIT_DEFIB_BLACKLISTED, REF(src))
```

Better trait source, the other one is meant for something else

File: code/modules/antagonists/revolution/revolution.dm [hunk 3]
```diff
@@ -441,39 +441,34 @@
...
-			if (!istype(target_body))
+			if (!istype(player))
 				continue
 
-			if (target_body.stat == DEAD)
-				target_body.makeUncloneable()
-			else
-				mind.announce_objectives()
+			if(player_mind.assigned_role.departments_bitflags & DEPARTMENT_BITFLAG_COMMAND)
+				ADD_TRAIT(player, TRAIT_DEFIB_BLACKLISTED, JOB_TRAIT)
```
Comment (Mothblocks) [inline code/modules/antagonists/revolution/revolution.dm:471]:
```suggestion
				ADD_TRAIT(player, TRAIT_DEFIB_BLACKLISTED, REF(src))
```

**Review by Mothblocks — CHANGES_REQUESTED** [review]
Can you also call whatever proc updates the med HUD, so they don't have a defib icon?

**Comment by Fikou** [comment]
let syndie defibs work on blacklisted people :trollface:

**Comment by JohnFulpWillard** [comment]
> let syndie defibs work on blacklisted people :trollface:

gonna use the syndie defib to revive the CMO

**Pull request merged**